app/src/main/java/org/jellyfin/androidtv/ui/playback/PlaybackController.java

```diff
@@ -719,12 +768,15 @@ public void switchAudioStream(int index) {
             return;
 
         int currAudioIndex = getAudioStreamIndex();
-        Timber.d("trying to switch audio stream from %s to %s", currAudioIndex, index);
-        if (currAudioIndex == index) {
-            Timber.d("skipping setting audio stream, already set to requested index %s", index);
-            if (mCurrentOptions.getAudioStreamIndex() == null || mCurrentOptions.getAudioStreamIndex() != index) {
-                Timber.d("setting mCurrentOptions audio stream index from %s to %s", mCurrentOptions.getAudioStreamIndex(), index);
-                mCurrentOptions.setAudioStreamIndex(index);
+        Integer currOptionsAudioIndex = mCurrentOptions.getAudioStreamIndex();
+        Integer prevOptionsAudioIndex = mPrevOptions.getAudioStreamIndex();
+        int audioIndexToUse = rememberPreviousAudioTrack && prevOptionsAudioIndex != null ? prevOptionsAudioIndex : index;
+        Timber.d("trying to switch audio stream from %s to %s", currAudioIndex, audioIndexToUse);
+        if (currAudioIndex == audioIndexToUse || (prevOptionsAudioIndex != null && prevOptionsAudioIndex == audioIndexToUse)) {
+            Timber.d("skipping setting audio stream, already set to requested index %s", audioIndexToUse);
+            if ((currOptionsAudioIndex == null || currOptionsAudioIndex != audioIndexToUse) || (prevOptionsAudioIndex == null || prevOptionsAudioIndex != audioIndexToUse)) {
+                Timber.d("setting mCurrentOptions audio stream index from %s to %s", currOptionsAudioIndex, audioIndexToUse);
+                mCurrentOptions.setAudioStreamIndex(audioIndexToUse);
             }
```

Comment:
It looks like you wanted these two conditions to be mutually exclusive. Current behavior would only attempt to change the audio stream index when the skip condition is true. That's probably what's leading to your infinite loop.
```suggestion
        if (currAudioIndex == audioIndexToUse || (prevOptionsAudioIndex != null && prevOptionsAudioIndex == audioIndexToUse)) {
            Timber.d("skipping setting audio stream, already set to requested index %s", audioIndexToUse);
        }
        else if ((currOptionsAudioIndex == null || currOptionsAudioIndex != audioIndexToUse) || (prevOptionsAudioIndex == null || prevOptionsAudioIndex != audioIndexToUse)) {
            Timber.d("setting mCurrentOptions audio stream index from %s to %s", currOptionsAudioIndex, audioIndexToUse);
            mCurrentOptions.setAudioStreamIndex(audioIndexToUse);
        }
```

Reply:
@ConnorS1110 forgot to tag